### PR TITLE
Remove regexp-opt workarounds for Emacs < 24

### PR DESCRIPTION
**Pull request by non-Jedi**
Additional simplification enabled by #87.

This turned into a bit of a general cleanup PR with the subsequent commits.

### CODE REVIEW

File: julia-mode.el [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ;;; julia-mode.el --- Major mode for editing Julia source code -*- lexical-binding: t -*-
 
-;; Copyright (C) 2009-2014 Julia contributors
-;; URL: https://github.com/JuliaLang/julia
+;; Copyright (C) 2009-2014 Julia contributors, 2015-2020 julia-mode contributors
+;; URL: https://github.com/JuliaEditorSupport/julia-emacs
 ;; Version: 0.4
 ;; Keywords: languages
+;; Package-Requires: ((emacs "24.3"))
```
Comment (tpapp) [inline julia-mode.el]: Do we actually need 24.3, or would 
```suggestion
;; Package-Requires: ((emacs "24"))
```
do?

File: julia-mode.el [hunk 2]
```diff
@@ -906,10 +883,6 @@ following commands are defined:
 
 (add-hook 'inferior-julia-mode-hook 'inferior-julia--initialize)
 
-;;;###autoload
-(defalias 'run-julia #'inferior-julia
-  "Run an inferior instance of `julia' inside Emacs.")
-
```
Comment (tpapp) [inline julia-mode.el]: I guess that we can remove this, but technically it is a breaking change, so we should mention it in the changelog.

File: julia-mode.el (hunk 1 above)
Comment (tpapp) [reply julia-mode.el]: Nevermind, I saw [this comment](https://github.com/JuliaEditorSupport/julia-emacs/pull/100/files#r386407333).

File: julia-mode.el (hunk 2 above)
Comment (non-Jedi) [reply julia-mode.el]: I'm actually going to add this back in. I did a quick survey of other external language packages with a REPL, and the `run-LANGUAGE` function seems like a convention (Haskell had it, and I think ess had it).

**Comment by non-Jedi** [comment]
I've force-pushed the last commit with `run-julia` added back in and the removal of `latexsub` added to the changelog.

File: julia-mode.el (hunk 2 above)
Comment (tpapp) [reply julia-mode.el]: Thanks for looking into this.

**Pull request merged**